addons/binding/org.openhab.binding.opendaikin/src/main/java/org/openhab/binding/opendaikin/internal/discovery/OpenDaikinACUnitDiscoveryService.java

```diff
@@ -0,0 +1,157 @@
+import org.eclipse.smarthome.core.net.NetUtil;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.opendaikin.internal.OpenDaikinBindingConstants;
+import org.openhab.binding.opendaikin.internal.OpenDaikinWebTargets;
+import org.openhab.binding.opendaikin.internal.config.OpenDaikinConfiguration;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Discovery service for Daikin AC units.
+ *
+ * @author Tim Waterhouse <tim@timwaterhouse.com> - Initial contribution
+ *
+ */
+@Component(service = DiscoveryService.class)
+public class OpenDaikinACUnitDiscoveryService extends AbstractDiscoveryService {
+    private static final String UDP_PACKET_CONTENTS = "DAIKIN_UDP/common/basic_info";
+    private static final int REMOTE_UDP_PORT = 30050;
+
+    private Logger logger = LoggerFactory.getLogger(OpenDaikinACUnitDiscoveryService.class);
+
+    private final Runnable scanner;
+    private ScheduledFuture<?> backgroundFuture;
+
+    private Client client = ClientBuilder.newClient();
+
+    public OpenDaikinACUnitDiscoveryService() {
+        super(Collections.singleton(OpenDaikinBindingConstants.THING_TYPE_AC_UNIT), 600, true);
+        scanner = createScanner();
+    }
+
+    @Override
+    protected void startScan() {
+        scheduler.execute(scanner);
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        logger.debug("Starting background discovery");
+
+        if (backgroundFuture != null && !backgroundFuture.isDone()) {
+            backgroundFuture.cancel(true);
+            backgroundFuture = null;
+        }
+        backgroundFuture = scheduler.scheduleWithFixedDelay(scanner, 0, 60, TimeUnit.SECONDS);
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        if (backgroundFuture != null && !backgroundFuture.isDone()) {
+            backgroundFuture.cancel(true);
+            backgroundFuture = null;
+        }
+
+        super.stopBackgroundDiscovery();
+    }
+
+    private Runnable createScanner() {
+        return () -> {
+            long timestampOfLastScan = getTimestampOfLastScan();
+            for (InetAddress broadcastAddress : getBroadcastAddresses()) {
+                logger.debug("Starting broadcast for {}", broadcastAddress.toString());
+
+                try (DatagramSocket socket = new DatagramSocket()) {
+                    socket.setBroadcast(true);
+                    socket.setReuseAddress(true);
+                    byte[] packetContents = UDP_PACKET_CONTENTS.getBytes(StandardCharsets.UTF_8);
+                    DatagramPacket packet = new DatagramPacket(packetContents, packetContents.length, broadcastAddress,
+                            REMOTE_UDP_PORT);
+
+                    // Send before listening in case the port isn't bound until here.
+                    socket.send(packet);
+
+                    while (receivePacketAndDiscover(socket)) {
```

Comment:
That could take a while? Is this really the proper way?

Reply:
Added a comment on this line. receivePacketAndDiscover will return false when it waits for a response for more than 1 second so this will terminate fairly quickly.

As for whether this is the proper way or not, I've found a lot of variety in the way broadcasting is done in addons, but they mostly all boil down to this pattern.